### DISCUSSION TITLE
Image Processing tab

### DISCUSSION THREAD

**Discussion by devilismyfriend** [hlky/stable-diffusion-webui#298] [Show and tell]
Hello,
Due to the need for more upscale options and the need to declutter the main interface I suggest the following change, I started coding it and will continue if there's approval for the concept

Main UI
<img width="1298" alt="image" src="https://user-images.githubusercontent.com/87043616/187346951-3d994cef-4be2-4d6f-9118-ef16f54191ae.png">

Proposed Changes:
-Removal of all upscaling options from the txt2img and the img2img tabs, including fix faces
-Creation of Image Processor tab
-User will be able to select procedures, currently only Fix Faces and Upscale are available
-Under Upscale modes the user will be able to choose different Upscaling options, labeled with speed expectations:
-Batch file selection for running the selected processors on a batch of images

Comparison between upscale modes:

Original 448x640
![image](https://user-images.githubusercontent.com/87043616/187347736-cc5ae769-1a74-4faa-8b6f-a35ccc62e646.png)

RealESRGAN, Fast, 4X, loss of details
![image](https://user-images.githubusercontent.com/87043616/187347816-a31ea87b-608e-461d-8976-5874fa0296d7.png)

GoBig, RealESRGAN 2X -> Sliced and SD diffused to add details, takes prompt/seed etc that will hopefully be pushable from the other tabs
![image](https://user-images.githubusercontent.com/87043616/187347860-62c0d9f0-4a9a-40fd-941d-0da7bc49ba03.png)

Latent Diffusion, Slow, 4X, doesn't alter the image like GoBig, produces some grain but is sharper and more accurate then RealESRGAN while providing better resolution then GoBig.
![image](https://user-images.githubusercontent.com/87043616/187347897-ec0bc99a-9cbf-4345-8b1d-8400faf90b88.png)

GoLatent, Slowest, 8X, combines the GoBig workflow with the Latent Diffusion upscaling method to add details and provide a clearer image through Latent Diffusion.
![image](https://i.imgur.com/sLvTLbI.jpg)

**Comment by LiJT** [comment]
Awesome!

**Comment by deegeeooh** [comment]
Looks great! Batch processing would be great! Also, saving of the upscaled images with the original prompt/filenames would be very handy! (now you have to manually save and rename..)

**Reply by devilismyfriend** [reply]
I agree, in any case I'm writing it for myself already, I refactored LDSR into its own module so it can be used easily from the script and I'm changing some of the webui code to load and unload the different systems as it's too memory hungry for my 3080 otherwise.

I'll post a link to my fork when it's ready if this doesn't pan out for this project.

**Comment by lashmanRB** [comment]
wow, that looks amazing, especially the GoLatent version! i really hope this gets added

and also really good idea with the separate tab, especially if we could move selected image from other tabs directly to it - that would be great

**Reply by devilismyfriend** [reply]
Yeah that's the idea, keep the generation and processing suite separate and have easy access to them

**Comment by JoshuaKimsey** [comment]
Wow, GoLatent looks amazing! Out of curiosity, does it require any added VRAM requirements than realESRGAN?

**Reply by devilismyfriend** [reply]
Yes, LDSR is a separate model, I actually have to unload SD first before upscaling with it to free up memory, so this feature will require at least 8-10GB VRAM.

**Reply by lashmanRB** [reply]
i would be absolutely fine with that. Actually, have you thought about making it available separately as well?

**Reply by devilismyfriend** [reply]
> i would be absolutely fine with that. Actually, have you thought about making it available separately as well?

I had to rewrite parts of the LDSR code because it was originally a collab/cog only thing, you'll have to sync down my repo "latent-diffusion" to use it on your own or on this project, the batch.py will process a set directory and the LDSR.py can be loaded as a module to other programs, I also included a download_model.bat file that will presumably download the model and config to the correct place.

**Reply by lashmanRB** [reply]
ohhh, that's great, thanks!!! :)

**Comment by devilismyfriend** [comment]
pull request is out for anyone that wants to test it
https://github.com/hlky/stable-diffusion-webui/pull/401

**Comment by yourjelly** [comment]
Could you edit to state the resolution on each of the examples like you do on the original for quick reference
